pkg/kubectl/cmd/scale.go

```diff
@@ -122,43 +121,35 @@ func RunScale(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []stri
 		return err
 	}
 
-	infos, err := r.Infos()
-	if err != nil {
-		return err
-	}
-	info := infos[0]
-	mapping := info.ResourceMapping()
-	scaler, err := f.Scaler(mapping)
-	if err != nil {
-		return err
-	}
+	counter := 0
+	err = r.Visit(func(info *resource.Info, err error) error {
+		if err != nil {
+			return err
+		}
 
-	resourceVersion := cmdutil.GetFlagString(cmd, "resource-version")
-	if len(resourceVersion) != 0 && len(infos) > 1 {
-		return fmt.Errorf("cannot use --resource-version with multiple resources")
-	}
-	currentSize := cmdutil.GetFlagInt(cmd, "current-replicas")
-	if currentSize != -1 && len(infos) > 1 {
-		return fmt.Errorf("cannot use --current-replicas with multiple resources")
```

Comment:
> and probably this one.

This one I could probably live with removing.  It's a little strange, but its not doomed to failure.